scripts/lib/docker-functions.sh

```diff
@@ -0,0 +1,409 @@
+        return 0
+    fi
+    echo "MAILTO=""\n$AUTO_BACKUP_INTERVAL cd /;/entrypoint.sh app:backup" > /etc/cron.d/autobackup
+    echo "Auto backup enabled."
+}
+run_initial_configuration() {
+    echo "=== Begin Initial Configuration Phase ==="
+    prepare_directories
+    configure_nginx
+    configure_certs
+    configure_database_settings
+    if [ "$MANUAL_CONFIGURATION" = "False" ] || [ "$MANUAL_CONFIGURATION" = "false" ]; then
+        configure_secrets
+        configure_authentication_backends
+        configure_zulip
+    fi
+    configure_auto_backup
+    echo "=== End Initial Configuration Phase ==="
+}
+# === bootstrap_environment ===
+wait_for_database() {
+    export PGPASSWORD="$DB_PASSWORD"
+    local TIMEOUT=60
+    echo "Waiting for database server to allow connections ..."
+    while ! /usr/bin/pg_isready -h "$DB_HOST" -p "$DB_HOST_PORT" -U "$DB_USER" -t 1 >/dev/null 2>&1
+    do
+        TIMEOUT=$(expr $TIMEOUT - 1)
+        if [[ $TIMEOUT -eq 0 ]]; then
+            echo "Could not connect to database server. Exiting."
+            unset PGPASSWORD
+            exit 1
+        fi
+        echo -n "."
+        sleep 1
+    done
+    unset PGPASSWORD
+}
+bootstrap_rabbitmq() {
+    echo "Bootstrapping RabbitMQ ..."
+    set +e
+    /root/zulip/scripts/setup/configure-rabbitmq | tail -n 16
+    RETURN_CODE=$?
+    if [[ $RETURN_CODE != 0 ]] && ([ "$IGNORE_RABBITMQ_ERRORS" = "False" ] || [ "$IGNORE_RABBITMQ_ERRORS" = "false" ]); then
+        echo "=> In most cases you can completely ignore the RabbmitMQ bootstrap errors."
+        echo "=> If you want to ignore RabbitMQ bootstrap errors, (re)add the env var 'IGNORE_RABBITMQ_ERRORS' with 'true'."
+        echo "Zulip RabbitMQ bootstrap failed in \"configure-rabbitmq\" exit code $RETURN_CODE. Exiting."
+        exit $RETURN_CODE
+    fi
+    set -e
+    echo "RabbitMQ bootstrap succeeded."
+}
+zulip_first_start_init() {
+    echo "Executing Zulip first start init ..."
+    if [ -e "$DATA_DIR/.initiated" ] && ([ "$FORCE_FIRST_START_INIT" != "True" ] && [ "$FORCE_FIRST_START_INIT" != "true" ]); then
+        echo "First Start Init not needed. Continuing."
+        return 0
+    fi
+    local RETURN_CODE=0
+    set +e
+
+    /home/zulip/deployments/current/scripts/setup/postgres-init-db
+    RETURN_CODE=$?
+    if [[ $RETURN_CODE != 0 ]]; then
+        echo "Zulip first start init failed at \"postgres-init-db\" with exit code $RETURN_CODE. Exiting."
+        exit $RETURN_CODE
+    fi
+
+    su zulip -c /home/zulip/deployments/current/scripts/setup/initialize-database
+    RETURN_CODE=$?
+    if [[ $RETURN_CODE != 0 ]]; then
+        echo "Zulip first start init failed at \"initialize-database\" with exit code $RETURN_CODE. Exiting."
+        exit $RETURN_CODE
+    fi
+
+    if ([ "$ZULIP_USER_CREATION_ENABLED" = "True" ] || [ "$ZULIP_USER_CREATION_ENABLED" = "true" ]); then
+        /home/zulip/deployments/current/scripts/create-zulip-admin
```

Comment:
I think the supervisord thing containing the create-zulip-admin can be removed given that it is already run here *once*.